ambari-logsearch-web/src/app/services/history-manager.service.ts

```diff
@@ -209,18 +190,14 @@ export class HistoryManagerService {
    * @returns {ListItem[]}
    */
   get redoItems(): ListItem[] {
-    const allItems = this.activeHistory.slice().reverse();
+    const allItems = [...this.activeHistory].reverse();
     let startIndex = allItems.findIndex((item: ListItem): boolean => {
-        return item.value.previousChangeId === this.currentHistoryItemId && !item.value.isUndoOrRedo;
-      }),
-      endIndex = allItems.slice(startIndex + 1).findIndex((item: ListItem): boolean => item.value.isUndoOrRedo);
+      return item.value.previousChangeId === this.currentHistoryItemId;
+    });
     if (startIndex === -1) {
       startIndex = allItems.length;
     }
-    if (endIndex === -1) {
```

Comment:
This isUndoOrRedo/endIndex logic shouldn't be removed here. It's needed to handle the situation when after some undoing/redoing user goes on with non-historical filters changes, so there's nothing to redo. Sorry for my omission in previous comment.
P.S. Getting rid of hasNoPendingUndoOrRedo looks OK to me.